src/plugins/intel_gpu/src/graph/half.cpp

```diff
@@ -1,165 +0,0 @@
-// Copyright (C) 2018-2023 Intel Corporation
-// SPDX-License-Identifier: Apache-2.0
-//
-
-#include <stdint.h>
-
-#ifdef HAVE_SSE
-#include <immintrin.h>
-#else
-#include "openvino/core/type/float16.hpp"
-#endif // HAVE_SSE
-
-#include "intel_gpu/runtime/half.hpp"
-
-namespace cldnn {
-
-#ifdef HAVE_SSE
-
-float half_to_float(uint16_t value) {
```

Comment:
does it mean an unoptimized conversion between float16 and float will be used?

Reply:
Yes, but it seems to be not worse than current half_to_float impl. I did a quick benchmark like this:
```cpp
    size_t count = 10000000;
    size_t repeats = 10;
    std::vector<ov::float16> float16_values(count);
    std::vector<uint16_t> half_values(count);
    std::vector<float> float_values_0(count);
    std::vector<float> float_values_1(count);

    for (size_t i = 0; i < count; i++) {
        random_generator rg("aaa");
        ov::float16 v = rg.generate_random_val<ov::float16>(-10, 10);
        half_values[i] = v.to_bits();
        float16_values[i] = v;
    }

    for (size_t r = 0; r < repeats; r++) {
        auto start = std::chrono::high_resolution_clock::now();
        for (size_t i = 0; i < count; i++) {
            float_values_0[i] = static_cast<float>(float16_values[i]);
        }
        auto end = std::chrono::high_resolution_clock::now();
        std::cerr << "float16: r = " << r << " time = " << std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() << std::endl;
    }

    for (size_t r = 0; r < repeats; r++) {
        auto start = std::chrono::high_resolution_clock::now();
        for (size_t i = 0; i < count; i++) {
            float_values_1[i] = half_to_float_(half_values[i]);
        }
        auto end = std::chrono::high_resolution_clock::now();
        std::cerr << "half_to_float: r = " << r << " time = " << std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() << std::endl;
    }
```

And got the following results:
```
float16: r = 0 time = 11535
float16: r = 1 time = 11556
float16: r = 2 time = 11038
float16: r = 3 time = 12377
float16: r = 4 time = 11183
float16: r = 5 time = 12224
float16: r = 6 time = 12508
float16: r = 7 time = 12559
float16: r = 8 time = 11675
float16: r = 9 time = 11949
half_to_float: r = 0 time = 13067
half_to_float: r = 1 time = 13257
half_to_float: r = 2 time = 12992
half_to_float: r = 3 time = 12881
half_to_float: r = 4 time = 13000
half_to_float: r = 5 time = 13026
half_to_float: r = 6 time = 12919
half_to_float: r = 7 time = 13033
half_to_float: r = 8 time = 12981
half_to_float: r = 9 time = 12993
```

> BTW, this code references half.cpp

Thanks, will remove that.